### PR TITLE
Added linters

### LINKED ISSUE #17: Run code linting in CI
We should run proper code linting on the code here, e.g., https://github.com/sunnysid3up/python-linter.

### PR DESCRIPTION
**Pull request by DunklesArchipel**
Fixes #17.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Lint
+on:
+  push: { branches: [ "master" ] }
+  pull_request: { branches: [ "master" ] }
+
```
Comment (saraedum) [inline .github/workflows/lint.yml:5]:
```suggestion
  push: { branches: [ "main" ] }
  pull_request: { branches: [ "main" ] }

```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with: { mamba-version: "*", channels: "conda-forge", channel-priority: true }
+      - name: install dependencies
+        shell: bash -l {0}
+        run: |
+          mamba env update --quiet -n test -f environment.yml
+          conda list
+      - name: pylint
+        shell: bash -l {0}
+        run: |
+          pylint echemdb
```
Comment (saraedum) [inline .github/workflows/lint.yml:26]: I am pretty sure that this will fail once it actually runs.

File: environment.yml [hunk 3]
```diff
@@ -4,8 +4,10 @@ channels:
   - defaults
 dependencies:
   - astropy
+  - black
   - filelock
   - inflect
+  - isort
```
Comment (saraedum) [inline environment.yml:10]: What about pylint?

**Review by saraedum — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (DunklesArchipel) [reply .github/workflows/lint.yml:26]: Indeed

File: environment.yml (hunk 3 above)
Comment (DunklesArchipel) [reply environment.yml:10]: I thought that black and isort were sort of same same but different. Now I know better when running pylint. :)

**Comment by DunklesArchipel** [comment]
This is what remains when legacy files are ignored
```
$ pylint echemdb --ignore-paths .+legacy.+
************* Module echemdb.website.filters.render
echemdb\website\filters\render.py:92:4: W0621: Redefining name 'render' from outer scope (line 63) (redefined-outer-name)
************* Module echemdb.website.filters
echemdb\website\filters\__init__.py:29:0: C0116: Missing function or method docstring (missing-function-docstring)
************* Module echemdb.website.generator.database
echemdb\website\generator\database.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module echemdb.website.generator.__main__
echemdb\website\generator\__main__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
echemdb\website\generator\__main__.py:36:0: C0116: Missing function or method docstring (missing-function-docstring)
echemdb\website\generator\__main__.py:40:13: C0103: Variable name "md" doesn't conform to snake_case naming style (invalid-name)
************* Module echemdb.website.macros.render
echemdb\website\macros\render.py:72:14: C0103: Argument name "f" doesn't conform to snake_case naming style (invalid-name)
************* Module echemdb.website.macros
echemdb\website\macros\__init__.py:27:0: C0116: Missing function or method docstring (missing-function-docstring)
************* Module echemdb.website.macros.__init__
echemdb\website\macros\__init__.py:1:0: R0401: Cyclic import (echemdb.website.filters -> echemdb.website.filters.render -> echemdb.website.macros.render) (cyclic-import)       
echemdb\website\macros\__init__.py:1:0: R0401: Cyclic import (echemdb.website.macros -> echemdb.website.macros.render) (cyclic-import)
```

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        with: { mamba-version: "*", channels: "conda-forge", channel-priority: true }
+      - name: install dependencies
+        shell: bash -l {0}
+        run: |
+          mamba env update --quiet -n test -f environment.yml
+          conda list
+      - name: pylint
+        shell: bash -l {0}
+        run: |
+          pylint echemdb --ignore-paths .+legacy.+
```
Comment (DunklesArchipel) [inline .github/workflows/lint.yml:26]: The ignored paths should probably be removed at some point.

Comment (saraedum) [reply .github/workflows/lint.yml:26]:
```suggestion
          pylint echemdb
```

Fixed in #147

**Review by saraedum — APPROVED** [review]

**Pull request merged**